tests/chain_swaps.rs

```diff
@@ -0,0 +1,477 @@
+                            create_chain_response.lockup_details.clone().lockup_address
+                        );
+                    }
+
+                    if update.status == "transaction.confirmed" {
+                        log::info!("Your lockup tx is confirmed!");
+
+                        std::thread::sleep(Duration::from_secs(10));
+                        log::info!("Claiming!");
+
+                        let claim_tx = LBtcSwapTxV2::new_claim(
+                            claim_script.clone(),
+                            claim_address.clone(),
+                            &ElectrumConfig::default_liquid(),
+                            BOLTZ_TESTNET_URL_V2.to_string(),
+                            swap_id.clone(),
+                        )
+                        .unwrap();
+                        let refund_tx = BtcSwapTxV2::new_refund(
+                            lockup_script.clone(),
+                            &refund_address,
+                            &ElectrumConfig::default_bitcoin(),
+                        )
+                        .unwrap();
+                        let claim_tx_response =
+                            boltz_api_v2.get_chain_claim_tx_details(&swap_id).unwrap();
+                        let (partial_sig, pub_nonce) = refund_tx
+                            .partial_sig(
+                                &our_refund_keys,
+                                &claim_tx_response.pub_nonce,
+                                &claim_tx_response.transaction_hash,
+                            )
+                            .unwrap();
+                        let tx = claim_tx
+                            .sign_claim(
+                                &our_claim_keys,
+                                &preimage,
+                                Amount::from_sat(1000),
+                                Some(Cooperative {
+                                    boltz_api: &boltz_api_v2,
+                                    swap_id: swap_id.clone(),
+                                    pub_nonce: Some(pub_nonce),
+                                    partial_sig: Some(partial_sig),
+                                }),
+                            )
+                            .unwrap();
+
+                        claim_tx
+                            .broadcast(&tx, &ElectrumConfig::default_liquid(), None)
+                            .unwrap();
+
+                        log::info!("Succesfully broadcasted claim tx!");
+                    }
+
+                    if update.status == "transaction.claimed" {
+                        log::info!("Successfully completed chain swap");
+                        break;
+                    }
+
+                    // This means the funding transaction was rejected by Boltz for whatever reason, and we need to get
+                    // fund back via refund.
+                    if update.status == "transaction.lockupFailed" {
```

Comment:
From experience (if you want to add this as a test) if you send less than the `create_chain_response.lockup_details.amount` you will get this error status